File: src/app/contracts/requests.py
from typing import Literal

from pydantic import BaseModel, Field

RequestType = Literal["feature", "bugfix", "refactor", "review", "test_plan"]
WriteAction = Literal["create_pr", "comment_pr", "update_issue", "apply_patch", "update_docs"]


class ArtifactRefs(BaseModel):
    issue_ids: list[str] = Field(default_factory=list)
    pr_url: str | None = None
    changed_files: list[str] = Field(default_factory=list)


class RequestOptions(BaseModel):
    include_tests: bool = True
    language: str = "ko"
    write_actions: list[WriteAction] = Field(default_factory=list)
    approval_token: str | None = None


class BaseWorkflowRequest(BaseModel):
    request_type: RequestType | None = None
    repo_id: str
    branch: str
    task_text: str
    artifacts: ArtifactRefs = Field(default_factory=ArtifactRefs)
    options: RequestOptions = Field(default_factory=RequestOptions)


class PlanRequest(BaseWorkflowRequest):
    pass


class ReviewRequest(BaseWorkflowRequest):
    diff_text: str | None = None


class TestPlanRequest(BaseModel):
    repo_id: str
    branch: str
    implementation_plan: list[str]
    impacted_areas: list[str]


class FeedbackRequest(BaseModel):
    run_id: str
    rating: int = Field(ge=1, le=5)
    useful: bool
    comment: str | None = None

File: frontend/src/api/contracts/requests.ts
export type RequestType = "feature" | "bugfix" | "refactor" | "review" | "test_plan";
export type WriteAction = "create_pr" | "comment_pr" | "update_issue" | "apply_patch" | "update_docs";

export interface ArtifactRefs {
  issue_ids: string[];
  pr_url: string | null;
  changed_files: string[];
}

export interface RequestOptions {
  include_tests: boolean;
  language: string;
  write_actions: WriteAction[];
  approval_token: string | null;
}

export interface BaseWorkflowRequest {
  request_type?: RequestType | null;
  repo_id: string;
  branch: string;
  task_text: string;
  artifacts: ArtifactRefs;
  options: RequestOptions;
}

export interface PlanRequest extends BaseWorkflowRequest {}

export interface ReviewRequest extends BaseWorkflowRequest {
  diff_text: string | null;
}

export interface TestPlanRequest {
  repo_id: string;
  branch: string;
  implementation_plan: string[];
  impacted_areas: string[];
}

export interface FeedbackRequest {
  run_id: string;
  rating: number;
  useful: boolean;
  comment: string | null;
}
